### PR TITLE
Canonicalize some user-provided paths

### PR DESCRIPTION
**Pull request by sellout**
Clearer paths to display to users and to avoid creating unused directories.

### CODE REVIEW

**Review by nuttycom — APPROVED** [review]
utACK. Since datadir changes are technically loss-of-funds potential, this should get one additional review.

File: src/util/system.cpp [hunk 1]
```diff
@@ -318,7 +318,7 @@ const fs::path GetExportDir()
     fs::path path;
     LOCK(cs_args);
     if (mapArgs.count("-exportdir")) {
-        path = fs::system_complete(mapArgs["-exportdir"]);
+        path = fs::weakly_canonical(fs::system_complete(mapArgs["-exportdir"]));
```
Comment (nuttycom) [inline src/util/system.cpp:321]: Can you also make the same change for `ZC_GetParamsDir`, and also check other path-related arguments that might appear in `mapArgs`?

**Review by nuttycom — CHANGES_REQUESTED** [review]
The same change could be usefully made in a few more places. Reviewed with @daira

**Review by str4d — APPROVED** [review]
utACK

File: src/util/system.cpp (hunk 1 above)
Comment (sellout) [reply src/util/system.cpp:321]: I made the change for `-paramsdir`, and couldn't find any other relevant paths.

**Review by nuttycom — APPROVED** [review]
utACK

**Pull request merged**